tests/funcadelic-test.js

```diff
@@ -52,6 +52,20 @@ describe('Semigroup', function() {
     }
     expect(append(new OneAndTwo(), { two: 'two', three: 3 })).to.be.instanceof(OneAndTwo);
   });
+  it('caches getters', () => {
+    let called = 0;
+    let result = append({ two: 2, three: 3 }, {
+      get sum() {
+        called++;
```

Comment:
Rather than testing for side-effects, what about just having:

```js
 let result = append({ two: 2, three: 3 }, {
   get sum() {
      return new Number(this.two + this.three);
   }
 }
```
and expect `result.sum === result.sum` ?
and expect `result.sum == 5`

Because what we really want to test for is stability, not how many times the getter was called.

Reply:
Yes, this is better. I'll change it.